_posts/2025-01-31-wildfly-domain-usage.adoc

```diff
@@ -0,0 +1,523 @@
+
+Now we can use the CLI to read the host name:
+
+[source,bash]
+----
+[domain@embedded /] /:read-attribute(name=local-host-name)
+{
+    "outcome" => "success",
+    "result" => "primary"
+}
+----
+
+It's preferred to use CLI command instead of directly operating on the host controller configuration file. In the rest of this blog post, I'll introduce both the content of the host controller file, and the way to use the CLI command to modify it.
+
+Now we can see the `host.xml` file has two servers settings by default:
+
+[source,xml]
+----
+<server name="server-one" group="main-server-group"/>
+<server name="server-two" group="main-server-group" auto-start="true">
+    <jvm name="default"/>
+    <socket-bindings port-offset="150"/>
+</server>
+<server name="server-three" group="other-server-group" auto-start="false">
+    <jvm name="default"/>
+    <socket-bindings port-offset="250"/>
+</server>
+----
+
+I’ll use only `server-three` in this blog post, so I comment out the `server-one` and the `server-two` definitions and only leave the `server-three` definition here.
+
+Here are the commands to disable the `server-one` and `server-two` from auto-starting:
+
+[source,bash]
+----
+[domain@embedded /] /host=primary/server-config=server-one:write-attribute(name=auto-start, value=false)
+{
+    "outcome" => "success",
+    "result" => undefined,
+    "server-groups" => undefined
+}
+
+[domain@embedded /] /host=primary/server-config=server-two:write-attribute(name=auto-start, value=false)
+{
+    "outcome" => "success",
+    "result" => undefined,
+    "server-groups" => undefined
+}
+
+[domain@embedded /]
+----
+
+Alternatively, you can use these commands to remove the `server-one` and `server-two` directly:
+
+[source,bash]
+----
+/host=primary/server-config=server-one:remove
+/host=primary/server-config=server-two:remove
+----
+
+In addition, there are several interface properties defined in the `host.xml` file that we need to override during runtime. Here is the `interfaces` section in the `host.xml`:
+
+[source,xml]
+----
+<interfaces>
+    <interface name="management">
+        <inet-address value="${jboss.bind.address.management:127.0.0.1}"/>
+    </interface>
+    <interface name="public">
+        <inet-address value="${jboss.bind.address:127.0.0.1}"/>
+    </interface>
+</interfaces>
+----
+
+I checked my IP address of Machine A, and it’s `192.168.0.115`, so I started the WildFly Server on Machine A by running the following commands in the `bin` directory of WildFly:
+
+[source,bash]
+----
+$ pwd
+/wildfly-35.0.0.Final/bin
+----
+
+[source,bash]
+----
+$ ./domain.sh --host-config=host.xml -Djboss.bind.address.management=192.168.0.115 -Djboss.bind.address=192.168.0.115 -Djboss.domain.primary.address=192.168.0.115
```

Comment:
Not critical and this can stay, but the `host.xml` is the default value for the `--host-config` argument. Meaning in this specific case, it's not required.

Reply:
add notes about the default value.